homeassistant/helpers/entity.py

```diff
@@ -92,7 +92,11 @@ def async_setup(hass: HomeAssistant) -> None:
 @bind_hass
 @singleton.singleton(DATA_ENTITY_SOURCE)
 def entity_sources(hass: HomeAssistant) -> dict[str, EntityInfo]:
-    """Get the entity sources."""
+    """Get the entity sources.
+
+    Items are added to this dict by Entity.async_internal_added_to_hass and
```

Comment:
I don't think we should have comments on place A saying that it is changed by B or C.
It usually makes it too easy for the comment to be stale and misleading in the future if the code changes. Without this, the developer just has to search for it or "find references", without getting mislead.

Why do you feel like this is needed? It seems easy too lookup already

Reply:
But now a wild PR appeared: https://github.com/home-assistant/core/pull/146549 🙈

Reply:
> I think it's a really weird pattern to have callers mutate a dict returned to them from a function, and it was not obvious to me that's how this is meant to be used. I would expect a bare dict in global scope instead.

True, it is weird indeed.

> You're right the comment is likely to grow stale though, maybe it can just mention callers of `helpers.entity.entity_sources` are expected to mutate the dict as needed?

Yeah, I think describing what/how it should be done instead of where it is done would make it better.

Reply:
This is merged now, but I still believe that it should be updated to something that will not get stale and be misleading too easily. It will become stale as soon as #146549 is merged